spyder/plugins/toolbar/container.py

```diff
@@ -0,0 +1,322 @@
+
+    # --- Public API
+    # ------------------------------------------------------------------------
+    def create_application_toolbar(self, toolbar_id, title):
+        """
+        Create a Spyder application toolbar and add it to the main window.
+
+        Paramaters
+        ----------
+        toolbar_id: str
+            The toolbar unique identifier string.
+        title: str
+            The localized toolbar title to be displayed.
+
+        Returns
+        -------
+        spyder.api.widgets.toolbar.ApplicationToolbar
+            The created application toolbar.
+        """
+        if toolbar_id in self._APPLICATION_TOOLBARS:
+            raise SpyderAPIError(
+                'Toolbar with id "{}" already added!'.format(toolbar_id))
+
+        toolbar = ApplicationToolBar(self, title)
+        toolbar.ID = toolbar_id
+        toolbar.setObjectName(toolbar_id)
+        self._APPLICATION_TOOLBARS[toolbar_id] = toolbar
+        self._toolbarslist.append(toolbar)
+
+        return toolbar
+
+    def add_application_toolbar(self, toolbar, mainwindow=None):
+        """
+        Add toolbar to application toolbars.
+
+        Parameters
+        ----------
+        toolbar: spyder.api.widgets.toolbars.ApplicationToolBar
+            The application toolbar to add to the `mainwindow`.
+        mainwindow: QMainWindow
+            The main application window.
+        """
+        toolbar_id = toolbar.ID
+        toolbar._check_interface()
+
+        if toolbar_id in self._ADDED_TOOLBARS:
+            raise SpyderAPIError(
+                'Toolbar with ID "{}" already added!'.format(toolbar_id))
+
+        # TODO: Make the icon size adjustable in Preferences later on.
+        iconsize = 24
+        toolbar.setIconSize(QSize(iconsize, iconsize))
+        toolbar.setObjectName(toolbar_id)
+
+        self._ADDED_TOOLBARS[toolbar_id] = toolbar
+        self._toolbarslist.append(toolbar)
+
+        if mainwindow:
+            mainwindow.addToolBar(toolbar)
+
+    def add_item_to_application_toolbar(self, item, toolbar, toolbar_id=None,
+                                        section=None, before=None):
+        """
+        Add action or widget `item` to given application menu `section`.
+
+        Parameters
+        ----------
+        item: SpyderAction or QWidget
+            The item to add to the `toolbar`.
+        toolbar: ApplicationToolBar or None
+            Instance of a Spyder application toolbar.
+        toolbar_id: str or None
+            The application toolbar unique string identifier.
+        section: str or None
+            The section id in which to insert the `item` on the `toolbar`.
+        before: str or None
+            Make the item appear before another given item.
+
+        Notes
+        -----
+        Must provide a `toolbar` or a `toolbar_id`.
+        """
+        if toolbar and toolbar_id:
+            raise SpyderAPIError('Must provide only toolbar or toolbar_id!')

```

Comment:
To make this work, `toolbar` should be optional too in the signature of this method.

Reply:
Fixed.